CppSource/Source/Services/Achievements/achievements.cpp

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pch.h"
+#include "xsapi/achievements_c.h"
+#include "achievements_helper.h"
+#include "achievements_taskargs.h"
+#include "xbox_live_context_impl.h"
+
+using namespace xbox::services;
+using namespace xbox::services::achievements;
+
+XSAPI_DLLEXPORT bool XBL_CALLING_CONV
+AchievementsResultHasNext(
+    _In_ XSAPI_ACHIEVEMENTS_RESULT* achievementsResult
+) XSAPI_NOEXCEPT
+try
+{
+    verify_global_init();
+
+    return achievementsResult->pImpl->cppAchievementsResult().has_next();
+}
+CATCH_RETURN()
+
+HC_RESULT get_next_execute(
+    _In_opt_ void* context,
+    _In_ HC_TASK_HANDLE taskHandle
+)
+{
+    auto args = reinterpret_cast<get_next_taskargs*>(context);
+    xbox_live_result<achievements_result> result = args->achievementsResult->pImpl->cppAchievementsResult().get_next(args->maxItems).get();
+    args->copy_xbox_live_result(result);
```

Comment:
it seems like the achievement result doesn't get passed back, only the error message